### ISSUE TITLE
Lean facade contract for stable webapp URL across releases

### ISSUE THREAD

**Issue by iduartgomez** [freenet/mail#200]
## Problem

Web-container contract ID rotates every release (#198) because Cargo.lock churn shifts wasm bytes. Inbox + AFT contract IDs also rotate (#199), orphaning per-identity state on chain. Users have no stable URL to bookmark, no recovery path for orphaned messages or token ledgers.

## Proposal

Introduce a **lean facade contract** with a permanent contract ID. Facade state is a small signed pointer to the current webapp contract ID. Releases update facade state, not the facade wasm. URL stays stable across releases; webapp bytes rotate behind it.

```
┌─────────────────────────────────────────────────┐
│  Facade contract  (STABLE id, never rebuilt)    │
│  state = { app_version, current_app_id, ... }   │
└─────────────────────────────────────────────────┘
                       │
                       │ points to
                       ▼
┌─────────────────────────────────────────────────┐
│  Current app contract  (ROTATES per release)    │
│  state = signed webapp bytes                    │
└─────────────────────────────────────────────────┘
```

### State (CBOR)

```
{
  version: u64,                        // monotonic; unix timestamp at signing
  current_app_id: [u8; 32],            // ContractInstanceId of webapp
  prev_app_ids: [(u64, [u8; 32])],     // last N for fallback (default N=3)
  signature: [u8; 64]                  // ed25519 over (version, current_app_id, prev_app_ids)
}
```

### Parameters

32-byte ed25519 verifying key (same prod key as today's web-container). Yields stable contract ID = `hash(facade_wasm, prod_pubkey)`. Permanent as long as `facade_wasm` + `prod_pubkey` are stable.

### update_state validation

- Decode CBOR.
- Verify ed25519 signature over `(version, current_app_id, prev_app_ids)`.
- Reject if `incoming.version <= current.version` (monotonic).
- Reject if `prev_app_ids.len() > N`.
- Otherwise replace state.

### summarize_state / get_state_delta

Standard freenet-stdlib boilerplate. State is small (<= ~512 bytes), so deltas can just be the full state.

## UI bootstrap flow

1. Browser opens `http://<gw>/v1/contract/web/<FACADE_ID>/`.
2. Facade contract serves a minimal loader HTML (or gateway shell wraps it).
3. Loader fetches `<FACADE_ID>` state via WS → reads `current_app_id`.
4. Loader navigates iframe to `http://<gw>/v1/contract/web/<current_app_id>/`.
5. Real app loads.

Loader is ~200 lines of vanilla JS. No Dioxus, no large deps. Lives in `contracts/facade-loader/` and is the only thing the facade contract serves.

## Build hygiene (critical)

The facade wasm MUST be byte-deterministic across all future builds. If it drifts even once, the contract ID rotates and the whole point is defeated.

Strategy:
1. Vendor prebuilt `facade.wasm` in `published-contract/facade.wasm`. Commit it.
2. CI step asserts that rebuilding the facade source produces the byte-identical wasm. If not, CI fails — release is gated on facade stability.
3. Pin facade's own `Cargo.lock` under `contracts/facade/Cargo.lock` (separate from workspace), with no path-deps that drag in workspace churn.
4. Facade source has zero deps beyond `freenet-stdlib` + `ed25519-dalek` + `ciborium`. No dioxus, no chrono, no anything that bumps frequently.
5. Document rustc version + target + flags required to reproduce the wasm. Pin in `rust-toolchain.toml` for the facade subdir if needed.

If we can't make rebuild deterministic, the fallback is: never rebuild. Treat the committed `facade.wasm` as the authoritative artifact.

## Release pipeline changes

```
Today:
  build → sign-webapp → update-published-contract → publish-email
                                                           │
                                                           └─→ new contract id every time

After:
  build → sign-webapp → publish webapp (rotating id is fine)
                              │
                              ▼
                        update facade state
                              │
                              └─→ facade id stays stable; current_app_id pointer flips
```

`scripts/release.sh` gains a final step: PUT facade state with new `current_app_id`. The facade contract itself is published once, then never re-published.

## Phasing

**Phase 1** — ship facade with manual `current_app_id` updates per release.
- URL stable.
- Inbox/AFT still rotate per #199.
- UI embeds `FACADE_ID` instead of `WEB_CONTAINER_CONTRACT_ID`.

**Phase 2** — auto-migration for inbox/AFT (#199 builds on this).

**Phase 3** — retire `published-contract/contract-id.txt` consumption from UI build. UI embeds only `FACADE_ID`. Release pipeline updates facade state, doesn't recompile UI to bake a new ID.

## Open questions

1. **Loader vs dispatcher contract.** Loader-in-HTML is simpler. Dispatcher contract (facade wasm reads `current_app_id` and returns those bytes directly) requires contract→contract calls — freenet-stdlib doesn't support that cleanly. Going with loader.
2. **Gateway caching.** Facade state changes need cache invalidation. May need a cache-bypass header on facade GETs, or a short TTL. Worst case: a release takes a minute to be visible to users.
3. **Rollback.** `prev_app_ids` lets UI fall back if `current_app_id` fails to load. UX: should rollback be automatic (try current, fall back on load error) or manual (toast: "new version failed to load, switch to previous?"). Lean toward automatic with a toast.
4. **Network propagation lag.** Facade UPDATE takes seconds-to-minutes to propagate. Users hitting facade right after release may see stale `current_app_id`. Acceptable.
5. **Key loss.** ed25519 key loss = facade unrecoverable. Same risk as today's prod key, but now with cascading impact since facade is the entry point. Mitigation: backup key offline (already required per RELEASING.md).

## Risks

- **Facade wasm drift.** Single point of failure. Mitigated by commit-prebuilt + CI byte-equality check.
- **Two contracts to track.** Operational complexity slightly up.
- **Browser cache pinning to old `current_app_id`.** If a user's browser caches the loader's iframe redirect, they may stay on the old app. Loader must use `no-cache` headers + force iframe reload on facade state change.

## Out of scope

- Decoupling URL from wasm hash at the freenet-core protocol level. That would obviate this issue but is a much larger change.
- Inbox/AFT migration (separate issue #199).
- Multi-key signing / governance for facade updates.

## Sub-tasks

- [ ] Design + implement `contracts/facade/` (state, params, validation).
- [ ] Design + implement `contracts/facade-loader/` (vanilla JS, ~200 lines).
- [ ] Pin facade build determinism (separate Cargo.lock, CI byte-equality check).
- [ ] Publish facade once to test net + production net; record IDs.
- [ ] UI: replace `WEB_CONTAINER_CONTRACT_ID` with `FACADE_ID`.
- [ ] Release pipeline: add facade-state UPDATE step at end of `scripts/release.sh`.
- [ ] Smoke test: release v0.x.0, then v0.x.1, verify same URL works.
- [ ] Document migration in AGENTS.md + RELEASING.md.

## Related

- #198 — web-container ID rotation root cause.
- #199 — inbox/AFT migration (orthogonal, both needed).
